folders.py

```diff
@@ -128,15 +129,22 @@ def get_folder(folder_id, data):
 
 def get_all_persistent_cache_files(path):
     """Get all files in PersistentCache storage with "start-group" marker."""
+    regex = ".*start-group.*"
+    regObj = re.compile(regex)
+    res = []
     path = os.path.expanduser(path)
-    shell_command = (
-        'grep -rls "start-group" "{path}" --null | xargs -0 ls -t'
-    ).format(path=path)
-    output = subprocess.check_output(shell_command, shell=True).strip()
-    if output:
-        return output.split(b'\n')
-    else:
-        return []
+    for root, dirs, fnames in os.walk(path):
+        for fname in fnames:
+            try:
+                with open(os.path.join(root, fname), encoding="cp437") as f:
+                    for line in f:
+                        if regObj.match(line):
```

Comment:
`b'start-group' in line` suffices here. No need for regex. (Syntax assumes binary mode.)

Reply:
For readability, you could also extract some of the lines into a function called `find(string, filepath)` which returns `True` or `False`.

Reply:
Lastly, if you see an easy way to make `PermissionError` backwards compatible to Python 2, that'll be great to. Right now, the code also works for Python 2. See also here: https://stackoverflow.com/a/18199529